### DISCUSSION TITLE
Using Evrete with Kotlin - unable to run the `prime numbers` example

### DISCUSSION THREAD

**Discussion by ramkumarkb** [evrete/evrete#97] [Q&A] [answered]
I am trying to use Evrete with Kotlin and trying the Prime number example - and I get the following warning - 

`org.evrete.runtime.AbstractRuleSession insertAtomic
WARNING: Can not map type for 'java.lang.Integer', insert operation skipped.`

In Kotlin - `Int` maps to java primitive `int` - tried couple of approached (e.g. using Kotlin's `IntArray` which maps Java int[] and using Int? which force Kotlin not to map to primite Java `int` - but no luck... 

Code (in Kotlin) - 

```
fun hello_world_rete() {
    val service = KnowledgeService()
    val knowledge = service
        .newKnowledge()
        .builder()
        .newRule("prime numbers")
        .forEach(
            "\$i1", Int::class.java,
            "\$i2", Int::class.java,
            "\$i3", Int::class.java
        )
        .where("\$i1 * \$i2 == \$i3")
        .execute { ctx: RhsContext -> ctx.deleteFact("\$i3") }
        .build()

    knowledge.newStatefulSession().use { session ->
        // Inject candidates
        val myList: List<Int?> = listOf<Int?>(2,3,4,5,6,7,8,9,10)
        val numbers = IntArray(100 - 2 + 1) { it + 2 }

//        for (i in 2..100) {
//        for (i in numbers) {
        for (i in myList) {
            session.insert(i)
        }

        // Execute rules
        session.fire()

        // Print current memory state
        session.forEachFact { handle: FactHandle?, o: Any? ->
            println(
                o
            )
        }
    }
    service.shutdown()
}
```

**Comment by ramkumarkb** [comment, marked as answer]
Ah! I think I understand the issue better - 
I have to use the Java `Integer` and not the Kotlin `Int` - 🤷 
```
        .forEach(
            "\$i1", Integer::class.java,
            "\$i2", Integer::class.java,
            "\$i3", Integer::class.java
        )
```   
Will continue digging Evrete with Kotlin... Thanks for the awesome library... !

**Comment by andbi** [comment]
Thanks for the answer. Using the library in Kotlin projects is an interesting topic. I'll look into it deeper this weekend.
